pkg/commands/resolver_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/google/ko/pkg/testutil"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	fooRef      = "github.com/awesomesauce/foo"
+	foo         = mustRandom()
+	fooHash     = mustDigest(foo)
+	barRef      = "github.com/awesomesauce/bar"
+	bar         = mustRandom()
+	barHash     = mustDigest(bar)
+	testBuilder = testutil.NewFixedBuild(map[string]v1.Image{
+		fooRef: foo,
+		barRef: bar,
+	})
+	testHashes = map[string]v1.Hash{
+		fooRef: fooHash,
+		barRef: barHash,
+	}
+)
+
+func TestResolveMultiDocumentYAMLs(t *testing.T) {
+	refs := []string{fooRef, barRef}
+	hashes := []v1.Hash{fooHash, barHash}
+	base := mustRepository("gcr.io/multi-pass")
+
+	buf := bytes.NewBuffer(nil)
+	encoder := yaml.NewEncoder(buf)
+	for _, input := range refs {
+		if err := encoder.Encode(input); err != nil {
+			t.Fatalf("Encode(%v) = %v", input, err)
+		}
+	}
+
+	inputYAML := buf.Bytes()
+
+	outYAML, err := resolveFile(
+		yamlToTmpFile(t, buf.Bytes()),
+		testBuilder,
+		testutil.NewFixedPublish(base, testHashes),
+		&options.SelectorOptions{},
+		&options.StrictOptions{})
+
+	if err != nil {
+		t.Fatalf("ImageReferences(%v) = %v", string(inputYAML), err)
+	}
+
+	buf = bytes.NewBuffer(outYAML)
+	decoder := yaml.NewDecoder(buf)
+	var outStructured []string
+	for {
+		var output string
+		if err := decoder.Decode(&output); err == nil {
+			outStructured = append(outStructured, output)
+		} else if err == io.EOF {
+			outStructured = append(outStructured, output)
+			break
+		} else {
+			t.Errorf("yaml.Unmarshal(%v) = %v", string(outYAML), err)
+		}
+	}
+
+	expectedStructured := []string{
+		testutil.ComputeDigest(base, refs[0], hashes[0]),
+		testutil.ComputeDigest(base, refs[1], hashes[1]),
+	}
+
+	// The multi-document output always seems to leave a trailing --- so we end up with
```

Comment:
This is a bug IMO

Reply:
This test is copied over - the extra `---` is for `kubectl` to trigger a flush I believe

Reply:
nm - looked at it. The test is using the decoder incorrectly.